lib/SearchProvider.js

```diff
@@ -52,7 +52,15 @@ NewTabSearchProvider.prototype = {
 
   get state() {
     return Task.spawn(function*() {
-      let state = yield ContentSearch._currentStateObj();
+      // a new version of ContentSearch.jsm landed in nightly but not firefox
+      // therefore, for now, we need to accomodate for both versions.
+      let state;
+      try {
+        state = yield ContentSearch.currentStateObj(true);
+      } catch (e) {
+        state = yield ContentSearch._currentStateObj();
+        Cu.reportError(e);
```

Comment:
This is expected behavior and we shouldn't report an error. Maybe it could be more explicit to do this as follows:

```javascript
let state;
try {
  if ("currentStateObj" in ContentSearch) {
    // ContentSearch.jsm from Firefox 49+
    state = yield ContentSearch.currentStateObj(true);
  } else {
    // legacy ContentSearch
    state = yield ContentSearch._currentStateObj();
  }
} catch(e) {
  Cu.reportError(e);
}
```